CUSTOMCONFIG.md

```diff
@@ -0,0 +1,402 @@
+     └── .d/
+         ├── 01-network.toml        # Additional network configuration
+         ├── 02-ssh.toml            # Additional SSH configuration
+         └── 99-overrides.toml      # Final overrides
+     ```
+
+#### 3. Defaults in Code
+   - If neither a file nor a directory is provided, `azure-init` falls back to using the default values specified in `config.rs`.
+
+---
+
+### Example 1: Single Configuration File
+**Command:**
+```sh
+azure-init --config /path/to/custom-config.toml
+```
+**Order of Merging:**
+1. Loads `/path/to/custom-config.toml`.
+2. Applies any CLI overrides if present.
+3. Fills in missing values with defaults from `config.rs`.
+
+### Example 2: Directory with Multiple .toml Files
+
+**Command:**
+```sh
+azure-init --config /path/to/custom-config-directory
+```
+**Directory Structure**
+```bash
+/path/to/custom-config-directory/
+├── azure-init.toml                # Base configuration
+└── .d/
+    ├── 01-network.toml            # Network configuration
+    ├── 02-ssh.toml                # SSH configuration
+    └── 99-overrides.toml          # Overrides
+```
+**Order of Merging:**
+
+1. Loads `azure-init.toml` as the base configuration.
+2. Merges `.d` files in alphabetical order:
+   - `01-network.toml`
+   - `02-ssh.toml`
+   - `99-overrides.toml`
+3. Applies any CLI overrides if present.
+4. Fills in missing values with defaults from `config.rs`.
+
+### Example 3: Default Path without --config
+**Assumption:** If no `--config` path is specified, `azure-init` will only use the default values set within `config.rs`.
+
+**Order of Loading:**
+
+1. Loads configuration directly from defaults specified in `config.rs`.
+
+## Validation Behaviors
+
+Custom enum types for `NetworkManager` and `SshAuthorizedKeysPathQueryMode` validate that config settings set by the user match the supported options. If a user enters an unsupported config value, deserialization will fail because Serde will not be able to map that value to one of the enum variants.
+
+```rust
+#[derive(Default, Serialize, Deserialize, Debug, Clone, Copy)]
+pub enum NetworkManager {
+    NetworkManager,
+    #[serde(rename = "systemd-networkd")]
+    #[default]
+    SystemdNetworkd,
+}
+
+#[derive(Default, Serialize, Deserialize, Debug, Clone, Copy)]
+pub enum SshAuthorizedKeysPathQueryMode {
+    #[serde(rename = "sshd -G")]
+    SshdG,
+    #[serde(rename = "disabled")]
+    #[default]
+    Disabled,
+}
+```
+
+#### Example of the expected error caused by entering an unsupported config value:
+
+```javascript
+Error: Invalid value 'unsupported_value' for NetworkManager. Expected 'NetworkManager' or 'systemd-networkd'.
+```
+
+The enum also relies on the `#[serde(rename = "disabled")]` attribute, which maps the Rust enum variant `Disabled` to the string "disabled."  This allows clear internal code naming while matching formats set by users in their configuration.
+
+### Configuration Fields
+
+#### Network Struct:
+- **manage_configuration**: `bool`
```

Comment:
are we managing network config? I don't think we currently have any network management code. I would not put it in and amend the documentation when we start influence networking behavior

Reply:
It was in the list of potential flags Chris and I had discussed but you're right -- we don't have any network management code at the moment so I will take it out until we do!